src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HexUtil.cs

```diff
@@ -23,6 +23,16 @@ internal static char IntToChar(uint i)
         }
 
         /// <summary>
+        /// Converts a number 0 - 15 to its associated hex character '0' - 'F'.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static char ToHexDigit(int value)
```

Comment:
This should sooooooo be a BCL method! As in a public BCL method on `System.Char`, either static or instance.